api/openapi-spec.yaml

```
@@ -0,0 +1,415 @@
+openapi: 3.0.1
+info:
+  title: Event Repository REST API
+  description: Event Repository REST API for retrieving the event information
+  contact: {}
+  version: 0.0.1
+tags:
+- name: API
+  description: Event Repository API
+paths:
+  /events:
+    get:
+      tags:
+      - API
+      summary: To get all events information
+      operationId: getEventsUsingGET
+      parameters:
+      - name: pageSize
+        in: query
+        description: "The number of events to be displayed per page."
+        schema:
+          type: integer
+          format: int32
+          default: 500
+      - name: params
+        in: query
+        description: |
+          To search for specific events or artifacts, filtering with parameters is supported.
+
+          **Syntax:**
+
+          `<resource>?key[.key ...]=value[&key[.key ...]=value ...]`
+
+          To traverse into nested structures and filter on their keys, namespacing with
+          `.` (dot) is used.
+
+          **Examples, single key:**
+
+          `/events?meta.type=EiffelActivityStartedEvent`
+
+          **Examples multiple keys:**
+
+          ```
+          /events?key1=value1&key2=value2
+          /events?key1=value1&key2=value2&key3=value3
+          ```
+
+          **Examples nested structures:**
+          ```
+          /events?data.identity=pkg:maven/my.namespace/my-name@1.0.0
+          /events?meta.source.domainId=my.domain&data.identity=pkg:maven/my.namespace/my-name@1.0.0    #Multiple keys and nested structures
+          ```
+
+          Note that mutiple keys only is allowed with logical AND (via `&`). There are no support for logical OR.
+
+          **Data types:**
+
+          By default, all data types are treated as strings. It is possible to specify the data type by including it explicitly in the query:
+
+          ```
+          /events?int(key1)>=value1&key2=value2&double(key3)=value3
+          ```
+
+          In this example, `key1` and `value1` are treated as integers, `key2`
+          and `value2` are treated as strings and `key3` and `values3` are
+          treated as doubles.
+
+          Currently only string (default), integer, long and double data types are supported.
+
+          **No comparator:**
+
+          To search for data that contains a field, use a query parameter
+          without comparator and value. For example `/events?key`fetches  all
+          events who has `key` as a field in the JSON documents. Here are some
+          examples:
+
+          ```
+          /events?data.identity                                   #Get all the events containing field 'identity' in the 'data'
+          /events?meta.source.domainId=my.domain&data.identity    #DomainId is 'my.domain' and has field 'identity' in 'data'
+          ```
+
+          **Example**
+
+          The following example adds two search parameters as two query keys (see https://swagger.io/docs/specification/serialization/ for more information).
```

I guess my confusion to this is based on that the actual example referred to is not part of this description, but rather in a separate `example` property in the `params` parameter. I suggest to rephrase this text to:
```suggestion
          The example provided in the `example` property of the `params` object adds two search parameters as two query keys (see https://swagger.io/docs/specification/serialization/ for more information).
```